resources/lemmas.md

```diff
@@ -214,6 +214,44 @@ These rules are specific to reasoning about EVM programs.
     rule N &Int MASK => N  requires MASK ==Int (2 ^Int (log2Int(MASK) +Int 1)) -Int 1 // MASK = 0xffff...f
                             andBool 0 <=Int N andBool N <=Int MASK
 
+
```

Comment:
No need for side consitions. We assume that elements in a WordStack are already in the required range. Also, if this rule has to stay, it should be made general for any WordStack Length. Look at the rules at the end of `verification.k` for example:
https://github.com/runtimeverification/verified-smart-contracts/blob/8b798a26f7adb30080b6b2aa13e050f36dc403ed/gnosis/verification.k#L75-L84

Reply:
if sizeofWordStack(W:WS) > 0,  rule nthbyteof(chop(#asWord(W : WS) &Int MASK), 0, N) will always rewrite to 0?

Reply:
No, why? `sizeofWordStack(W:WS)` is of course always > 0. And `((MASK >>Byte #sizeWordStack(WS)) &Int 255)` just extracts the byte on the right position from MASK. The piece `&Int 255` at the end is actually redundant because W is always < 255 and upper bytes of the mask won't affect the result.